File: 2023/domaci/47-adresarovy-strom/volfmatous/main.py
import os
import sys


def print_directory_contents(folder_path: str, nest_level: int = 0) -> None:
    files = sorted(os.listdir(folder_path))
    directories = sorted([d for d in files if os.path.isdir(os.path.join(folder_path, d))])
    files = sorted([f for f in files if os.path.isfile(os.path.join(folder_path, f))])

    print(" " * max(nest_level - 1, 0) * 2, end="")
    if nest_level > 0:
        print("- ", end="")
    print(os.path.basename(folder_path))

    for file in files:
        print(" " * nest_level * 2 + f"- {file}")

    for directory in directories:
        print_directory_contents(os.path.join(folder_path, directory), nest_level + 1)


if __name__ == "__main__":
    folder_path = input()

    if not os.path.exists(folder_path):
        print("not found")
        sys.exit()

    print_directory_contents(folder_path)

File: 2023/domaci/47-adresarovy-strom/volfmatous/main.cpp
#include <iostream>
#include <filesystem>
#include <algorithm>

namespace fs = std::filesystem;

void printDirectoryContents(const std::string& folderPath, int nestLevel) {
    std::vector<std::string> files, directories;

    for (const auto& entry : fs::directory_iterator(folderPath)) {
        if (entry.is_regular_file())
            files.push_back(entry.path().filename().string());
        else if (entry.is_directory())
            directories.push_back(entry.path().filename().string());
    }

    std::sort(files.begin(), files.end());
    std::sort(directories.begin(), directories.end());

    std::cout << std::string(std::max(nestLevel - 1, 0) * 2, ' ');
    if (nestLevel > 0) {
        std::cout << "- ";
    }
    std::cout << fs::path(folderPath).filename().string() << std::endl;

    for (const auto& file : files) {
        std::cout << std::string(nestLevel * 2, ' ') << "- " << file << std::endl;
    }

    for (const auto& directory : directories) {
        printDirectoryContents(folderPath + "/" + directory, nestLevel + 1);
    }
}

int main() {
    std::string folderPath;
    std::getline(std::cin, folderPath);

    if (!fs::exists(folderPath)) {
        std::cout << "not found" << std::endl;
        return 0;
    }

    printDirectoryContents(folderPath, 0);
    return 0;
}
